File: app/src/main/java/social/entourage/android/user/role/BaseUserRolesFactory.java
package social.entourage.android.user.role;

import java.util.ArrayList;

/**
 * Factory for user roles
 * Created by Mihai Ionescu on 21/05/2018.
 */
public class BaseUserRolesFactory {

    // ----------------------------------
    // ATTRIBUTES
    // ----------------------------------

    private ArrayList<UserRole> userRoles = new ArrayList<>();

    // ----------------------------------
    // PUBLIC METHODS
    // ----------------------------------

    public void register(UserRole userRole) {
        userRoles.add(userRole);
    }

    public UserRole findByName(String name) {
        if (name == null) return null;
        for (UserRole userRole : userRoles) {
            if (name.equalsIgnoreCase(userRole.getName())) {
                return userRole;
            }
        }
        return null;
    }

}

File: app/src/main/java/social/entourage/android/user/role/BaseUserRolesFactory.kt
package social.entourage.android.user.role

import java.util.*

/**
 * Factory for user roles
 * Created by Mihai Ionescu on 21/05/2018.
 */
open class BaseUserRolesFactory {
    // ----------------------------------
    // ATTRIBUTES
    // ----------------------------------
    private val userRoles = ArrayList<UserRole>()

    // ----------------------------------
    // PUBLIC METHODS
    // ----------------------------------
    fun register(userRole: UserRole) {
        userRoles.add(userRole)
    }

    fun findByName(name: String): UserRole? {
        for (userRole in userRoles) {
            if (name.equals(userRole.name, ignoreCase = true)) {
                return userRole
            }
        }
        return null
    }
}